pandas/_libs/lib.pyx

```diff
@@ -2669,6 +2678,20 @@ def maybe_convert_objects(ndarray[object] objects,
             return pi._data
         seen.object_ = True
 
+    elif seen.str_:
+        if is_string_array(objects):
```

Comment:
I know everywhere else does this, but is there a way to avoid this double parsing?

(Maybe we check the other flags are all false?)

Reply:
No, you exit the first loop as soon as you find one string